examples/TenStage/TenStage.bsv

```diff
@@ -0,0 +1,91 @@
+// a ten stage pipeline 
+// add 1 to the input in each stage
+// will stuck at stage5 if data is 32'h5c
+
+import RProbe::*;
+import FIFOF::*;
+import Vector::*;
+
+(* synthesize *)
+module mkAdderPipeline(Empty);
+    function gen_fire_probes(Integer x) = mkRProbe(fromInteger(x + 10));
```

Comment:
by the way, is it possible to use a string as the probe id to make it more meaningful?